src/pybind/mgr/dashboard/controllers/rbd.py

```diff
@@ -186,15 +182,20 @@ def _rbd_image(cls, ioctx, pool_name, image_name):
     def _rbd_pool_list(cls, pool_name):
         rbd_inst = rbd.RBD()
         with mgr.rados.open_ioctx(pool_name) as ioctx:
-            names = rbd_inst.list(ioctx)
             result = []
-            for name in names:
-                try:
-                    stat = cls._rbd_image(ioctx, pool_name, name)
-                except rbd.ImageNotFound:
-                    # may have been removed in the meanwhile
-                    continue
-                result.append(stat)
+            namespaces = rbd_inst.namespace_list(ioctx)
+            # images without namespace
+            namespaces.append('')
+            for namespace in namespaces:
+                ioctx.set_namespace(namespace)
+                names = rbd_inst.list(ioctx)
+                for name in names:
+                    try:
+                        stat = cls._rbd_image(ioctx, pool_name, namespace, name)
+                    except rbd.ImageNotFound:
+                        # may have been removed in the meanwhile
+                        continue
+                    result.append(stat)
             return result
```

Comment:
@epuertat please take a look into this commit https://github.com/ricardoasmarques/ceph/commit/fb1baa6648b35c242fbf37079cf5433300c4a2a7 and let me know if this addresses this comment, so I can rebase it.

The only downside I see here is that we are now opening more "ioctx" connections to the cluster, but the benefit of being able to cache those values in the future will justify that.

Reply:
I share your concern regarding the ioctx. We may temporarily pass the context too as an argument, and once the caching is in place we may easily remove it from the function interface. Thanks for the refactoring!

Reply:
Now the fix looks like this: https://github.com/ricardoasmarques/ceph/commit/e936d788b40089edcb11dc416925ef539b7e5ac3

Passing the "ioctx" context as an argument forced me to not use the "set_namespace" inside those functions to avoid side effects, which forced me to change the signature.

Those functions look a bit useless now but having those separate functions may help in the future, when implementing the cache. Let me know if it looks ok for rebase now.

Reply:
Thanks! As it should be transitory issue, no problem with that.

Reply:
Fixed